### PR TITLE
feat: Add Pool ACL tests (SC-403)

### PR DESCRIPTION
**Pull request by lucas-manuel**

### CODE REVIEW

**Review by barrutko — CHANGES_REQUESTED** [review]
Not sure if that falls into the scope of that PR, but I miss sanity checks that the state actually changes when called by a proper address. I.e. after successfully calling `configureEModeCategory` I'd expect a simple sanity check that the config of a given EMode category changed. But maybe this should be tested in PoolConfigurator and here testing exclusively the access is ok. Im'm open to discussion.

**Comment by lucas-manuel** [comment]
@barrutko Yeah I opted to do ACL testing first here in isolation to make sure that its covered across the whole protocol. The state assertions for non-setter ACL functions would've made the scope too large so I decided to put them all here first. Open to discussion as well, just thought this would be a good measure for us to take to make sure our bases were covered with this critical functionality first.

**Review by barrutko — DISMISSED** [review]
Yeah, this sounds reasonable to me, as long these end up being tested somewhere sometime anyway.

Smaller PR = ❤️

File: test/ACL.t.sol [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+        // Routes to fallback which EVM reverts when selector doesn't match on pool implementation
+        vm.expectRevert(bytes(""));
+        poolProxy.upgradeTo(address(borrowAsset));  // Use an address with code
+
+        vm.prank(POOL_ADDRESSES_PROVIDER);
+        poolProxy.upgradeTo(address(borrowAsset));  // Use an address with code
+    }
+
+    // NOTE: This function signature does NOT match what's on mainnet for the Pool proxy.
+    // TODO: Investigate this.
```
Comment (hexonaut) [inline test/ACL.t.sol:198]: What is meant by this? The `upgradeToAndCall` works fine in the test code, and it appears in both the deployed proxy and repo.

**Review by barrutko — APPROVED** [review]

**Review by hexonaut — APPROVED** [review]

**Pull request merged**